src/main/java/org/jabref/logic/citationstyle/CitationStyle.java

```diff
@@ -138,25 +139,26 @@ public static CitationStyle getDefault() {
             return STYLES;
         }
 
-        URL url = CitationStyle.class.getResource(STYLES_ROOT);
+        URL url = CitationStyle.class.getResource("/modules/org.jabref/csl-styles");
```

Comment:
I guess the problem is that "csl-styles" is outside of the module boundaries. Can you try to load a) a single csl file and b) see if it works if you copy the file to "src\main\java\org\jabref\logic\citationstyle" and load it via `CitationStyle.class.getResource("file.csl");` and finally c) if you place the file in the resource folder "src\main\resources\org\jabref\logic\citationstyle"

Reply:
Your guess was right. I copied over the files to src\main\java\org\jabref\logic\citationstyle and then did CitationStyle.class.getResource("file.csl"); and I got a valid url back which was non null.
Will try to test further